edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/ClientProvider.cs

```diff
@@ -18,12 +18,12 @@ public IClient Create(IIdentity identity, IAuthenticationMethod authenticationMe
 
             if (identity is IModuleIdentity)
             {
-                ModuleClient moduleClient = ModuleClient.Create(identity.IotHubHostName, authenticationMethod, transportSettings);
+                ModuleClient moduleClient = ModuleClient.Create(identity.IotHubHostname, identity.GatewayHostname.OrDefault(), authenticationMethod, transportSettings);
```

Comment:
Using OrDefault is not a good idea. You can map/match on the identity.GatewayHostname